python/tests/core/constraints/test_helpers.py

```diff
@@ -0,0 +1,205 @@
+import pytest
+
+from whylogs.core.constraints import (
+    ConstraintsBuilder,
+    count_below_number,
+    distinct_number_in_range,
+    frequent_strings_in_reference_set,
+    greater_than_number,
+    mean_between_range,
+    n_most_common_items_in_set,
+    null_percentage_below_number,
+    null_values_below_number,
+    quantile_between_range,
+    smaller_than_number,
+    stddev_between_range,
+)
+
+
+@pytest.fixture
+def builder(profile_view):
```

Comment:
Can we add a test against a NaN value?

Reply:
you mean having a profile_view column with only NaNs and running all tests against it? or just insert some NaNs in the column itself

Reply:
something like:
```python
@pytest.fixture
def nan_builder():
    df = pd.DataFrame([np.nan, np.nan], columns=["a"])
    profile_view = why.log(pandas=df).view()
    nan_builder = ConstraintsBuilder(dataset_profile_view=profile_view)
    return nan_builder
 
 def test_greater_than_number_with_nan(nan_builder):
    nan_builder.add_constraint(greater_than_number(column_name="a", number=2.2))
    constraint = nan_builder.build()

    assert constraint.validate() is False
```

Reply:
otherwise the only difference would be to always use the `weight` column, since it contains a NaN ;)

Reply:
Right, your nan_builder looks good, as long as there is one NaN value in our test input considering we had seen pitfalls around that scenario. Maybe one bad row, or an empty dataset kind of thing is a realistic scenario if the inputs come in batches over time, we can test that the constraints don't fail when there is a NaN value in there.

The other idea was to profile an empty numeric column so that the DistributionMetric contained NaN for the max/min  so we can show that the constraint condition wasn't surprising in the results. I'm expected we want to skip failing when evaluating and empty distribution metric with respect to a `greater_than_number ` constraint condition?

Reply:
for the second idea, we can make that optional, as @FelipeAdachi did for `quantile_between_range`. should that be the case for all  metrics then?

Reply:
its normal to have empty batches of data in some window of time if we are considering streaming scenarios or segments, and I don't think of those cases where there is no data as violating a range based constraint. Having an optional parameter is fine to control skipping NaN.